server/services/authorization/canAccessRoom.ts

```diff
@@ -55,7 +74,16 @@ const roomAccessValidators: RoomAccessValidator[] = [
 		if (!room?._id || !user?._id) {
 			return false;
 		}
-		if (await Subscriptions.countByRoomIdAndUserId(room._id, user._id)) {
+
+		const hasViewPermission = await roomTypePermission(room.t, user._id);
+
+		const hasSubscriptions = await Subscriptions.countByRoomIdAndUserId(room._id, user._id);
+
+		if (!hasViewPermission) {
+			return false;
```

Comment:
this will not block the access, it will only ignore the subscriptions.

Reply:
I did some changes and this [commit](https://github.com/RocketChat/Rocket.Chat/pull/24534/commits/b14482faa69e16143eff915b658a0cad44f572b5) can be the best approach.